src/Hooks/UtilityHooks.res

```diff
@@ -16,9 +16,13 @@ let useHandlePostMessages = (~complete, ~empty, ~paymentType, ~savedMethod=false
   let loggerState = Recoil.useRecoilValueFromAtom(loggerAtom)
   let setIsPayNowButtonDisable = Recoil.useSetRecoilState(payNowButtonDisable)
   let {sdkHandleConfirmPayment} = Recoil.useRecoilValueFromAtom(optionAtom)
+  let showMainScreen = Recoil.useRecoilValueFromAtom(showCardFieldsAtom)
 
   React.useEffect(() => {
-    if !sdkHandleConfirmPayment.allowButtonBeforeValidation {
+    let isPaymentMethodScreenAndWallet =
+      showMainScreen &&
+      (paymentType === "google_pay" || paymentType === "paypal" || paymentType === "apple_pay")
```

Comment:
yes it will break in case when google_pay or apple_pay is displayed in tabs flow, for handling that case we need to remove this check and will only call this `useHandlePostMessages` hook when google_pay or apple_pay are in the wallets flow. will do the changes for the same